scripts/ecs-anywhere-install.sh

```diff
@@ -0,0 +1,526 @@
+            fail
+        fi
+    elif [ "$(systemctl is-enabled amazon-ssm-agent)" == "enabled" ]; then
+        echo "ssm agent is installed, checking if the instance is registered."
+        check-instance-registered
+        if $INSTANCE_REGISTERED; then
+            echo "Instance has been registered."
+            return
+        else
+            echo "SSM Agent is installed. The instance needs to be registered."
+            register-ssm-agent
+            return
+        fi
+    else
+        local dir
+        dir="$(mktemp -d)"
+        ssm-agent-signature-verify
+        case "$PKG_MANAGER" in
+        dnf)
+            dnf install -y "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/linux_$ARCH_ALT/amazon-ssm-agent.rpm"
+            ;;
+        yum)
+            yum install -y "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/linux_$ARCH_ALT/amazon-ssm-agent.rpm"
+            ;;
+        apt)
+            dpkg -i "$dir/ssm-agent.deb"
+            ;;
+        zypper)
+            rpm --install "$dir/ssm-agent.rpm"
+            ;;
+        esac
+        rm -rf "$dir"
+    fi
+    # register the instance
+    register-ssm-agent
+
+    ok
+}
+
+ssm-agent-signature-verify() {
+    echo "Verifying the signature of SSM Agent"
+
+    # TODO [Update before release] Change this url to main repo master branch
+    curl -o "$dir/amazon-ssm-agent.gpg" "https://github.com/Realmonia/amazon-ecs-init/blob/ssmGpg/scripts/amazon-ssm-agent.gpg"
+    gpg --import "$dir/amazon-ssm-agent.gpg" &> "$dir/tempOut"
+    IFS=' '
+    read -ra ARR <<< "$(cat "$dir/tempOut")"
+    fingerprint=$(gpg --fingerprint "${ARR[2]%?}" | sed -n 2p | sed 's/ *$//g')
+    if [ "${fingerprint//[[:blank:]]/}" != "8108A07A9EBE248E3F1C63F254F4F56E693ECA21" ]; then
+        echo "SSM Agent GPG public key fingerprint verification fail. Stop the installation of the SSM Agent. Please contact AWS Support."
+        fail
+    fi
+    rm "$dir/tempOut"
+
+    case "$PKG_MANAGER" in
+    dnf | yum | zypper)
+        curl -o "$dir/ssm-agent.rpm" "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/linux_$ARCH_ALT/amazon-ssm-agent.rpm"
+        curl -o "$dir/ssm-agent.rpm.sig" "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/linux_$ARCH_ALT/amazon-ssm-agent.rpm.sig"
+        gpg --verify "$dir/ssm-agent.rpm.sig" "$dir/ssm-agent.rpm" &> "$dir/verifyOut"
+        ;;
+    apt)
+        curl -o "$dir/ssm-agent.deb" "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/debian_$ARCH_ALT/amazon-ssm-agent.deb"
+        curl -o "$dir/ssm-agent.deb.sig" "https://s3.$REGION.amazonaws.com/amazon-ssm-$REGION/latest/debian_$ARCH_ALT/amazon-ssm-agent.deb.sig"
+        gpg --verify "$dir/ssm-agent.deb.sig" "$dir/ssm-agent.deb" &> "$dir/verifyOut"
+        ;;
+    esac
+
+    if grep -Fxq "BAD signature" "$dir/verifyOut"; then
```

Comment:
instead of relying on the output going to a file and grepping for an error, I think it would be better to just check if the `gpg` command errors directly.

ie, this if-statement could look something like (probably changing the name of the deb/rpm file to a variable so we only have to have it once):

```
if gpg --verify "$dir/ssm-agent.deb.sig" "$dir/ssm-agent.deb"; then
    echo SUCCESS
else
    echo UH OH
fi
```

Reply:
Updated.